cirq/contrib/routing/greedy.py

```diff
@@ -0,0 +1,283 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import random
+from typing import (Any, Callable, cast, Dict, Hashable, Iterable, List,
+                    Mapping, Optional, Sequence, Set, Tuple)
+
+import numpy as np
+import networkx as nx
+
+from cirq import circuits, ops
+import cirq.contrib.acquaintance as cca
+from cirq.contrib.routing.initialization import get_initial_mapping
+from cirq.contrib.routing.swap_network import SwapNetwork
+from cirq.contrib.routing.utils import (get_timeslices,
+                                        are_ops_consistent_with_device_graph)
+
+SWAP = cca.SwapPermutationGate()
+
+
+def route_circuit_greedily(circuit: circuits.Circuit, device_graph: nx.Graph,
+                           **kwargs) -> SwapNetwork:
+    """Greedily routes a circuit on a given device.
```

Comment:
Please add a docstring (on top of the one-liner) that gives a description of the greedy strategy.

Reply:
Document the exposed function instead of the hidden class.

Reply:
Done